inst/base/mrgsolve-tad.h

```diff
@@ -0,0 +1,109 @@
+
+
+namespace mrgsolve {
+/** 
+ * Caculate time after dose for a specific dosing compartment. 
+ * 
+ * NOTE: this is in the `mrgsolve` namespace, so use `mrg::tadose` to construct
+ * these objects. You can include the header file or use the `tad` plugin (see
+ * examples below).
+ * 
+ * First, create a `tadose` object and assign a compartment number. The 
+ * compartment number can be assigned in the constructor or by assigning to the 
+ * `cmt` member. Calling the `tad()` method will calculate time after dose 
+ * for the designated compartment. The value `-1` is returned for records that 
+ * occur prior to the first administered dose.
+ * 
+ * There are a couple of approaches.  
+ * 
+ * First, you could create static objects in `[main]`. This code would give 
+ * you a global variable called `time_after_dose` and also includes the result
+ * in the simulated output.
+ * 
+ * @code
+ * [ plugin ] tad
+ * 
+ * [ main ] 
+ * static mrg::tadose obj(1); 
+ * 
+ * capture time_after_dose = obj.tad(self);
+ * @endcode 
+ * 
+ * The other way would be to declare the object globally in `[global]`.  For 
+ * example:
+ * 
+ * @code
+ * [ plugin ] tad
+ * 
+ * [ global ] 
+ * mrg::tadose obj;
+ * 
+ * [ preamble ] 
+ * obj.cmt = 2;
+ * 
+ * [ main ] 
+ * capture time_after_dose = obj.tad(self);
+ * @endcode
+ * 
+ */
+class tadose {
+public: 
+  int cmt; ///< dosing compartment number to follow 
+  double told; ///< time of last dose
+  bool had_dose; ///< has the current individual received a dose yet?
+  tadose(int cmt_);
+  tadose();
+  double tad(databox& self);
+  void reset();
+};
+/**
+ * Constructor.
+ * 
+ * @param cmt_ compartment number
+ * 
+ */
+tadose::tadose(int cmt_) {
```

Comment:
In general it's not a good idea to leave things uninitialized. Above all it simply makes it hard to debug. So for `cmt` we can give it a large number, such as `std::numeric_limits<int>`. For `had_dose` we need to set it to `false`.

Reply:
I should just include a comment like this in a PR review template so you don't have to type it out every single review :). 

1. Providing default values for both `cmt` and `had_dose`
1. Including default initialization in the test model